lib/tasks/repository.rake

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+namespace :repository do
+  desc "Load all Clients into Reference Repostories"
+  task load_client_repos: :environment do
+    puts "Processing Client Repositories"
+    progressbar = ProgressBar.create(
+      format: "%a %e %P% Processed: %c from %C %t",
+      title: "Client Repositories",
+      total: Client.all.count
+    )
+    Client.all.each do |c|
+      progressbar.increment
+      ReferenceRepository.find_or_create_by(
+        client_id: c.symbol,
+        re3doi: c.re3data_id
+      )
+    end
+  end
+
+  desc "Load all Re3data Repositories into Reference Repostories"
+  task :load_re3data_repos, [:pages] => :environment do |t, args|
```

Comment:
I'd move the actual fetching from re3data and such into the model, then this rake task calls that.
That way you can open the import of re3data (or datacite client) code to a background job.
i.e. for say updating based on when one of our clients gets updated.

Reply:
This is for initial load much like a db:seed.  All of the indexing and updating for individuals can be done bone in background jobs.